### ISSUE TITLE
RemovedInPytest4Warning: MarkInfo objects are deprecated

### ISSUE THREAD

**Issue by sliwinski-milosz** [pytest-dev/pytest-bdd#261]
```
/usr/local/lib/python3.5/site-packages/pytest_bdd/reporting.py:78: RemovedInPytest4Warning: MarkInfo objects are deprecated as they contain merged marks which are hard to deal with correctly.
Please use node.get_closest_marker(name) or node.iter_markers(name).
Docs: https://docs.pytest.org/en/latest/mark.html#updating-code
  param_names = parametrize.args[0] if isinstance(parametrize.args[0], (tuple, list)) else [
```

Line that invokes warning:
https://github.com/pytest-dev/pytest-bdd/blob/master/pytest_bdd/reporting.py#L76

Replacing below (as suggested in warning message):
```
parametrize = node.keywords._markers.get('parametrize')
```
with:
```
node.get_marker('parametrize')
```
makes tests still pass and we get rid of warning however I need to dig deeper into reporting logic to understand if above fix don't break anything. `get_closest_marker` works too.

Tested on:
- `pytest-bdd==3.0.0`
- `pytest==3.9.2`

EDIT:
`get_marker` also triggers a warning as it is deprecated too. `get_closest_marker` does not trigger it.

**Comment by youtux** [comment]
It seems to me that Node.get_marker is deprecated since pytest 3.6. I find it strange that using node.get_marker('parametrize') and then accessing parametrize.args doesn't trigger the warning.
I would just use get_closes_marker, to be honest. We have tests for reporting, so the feature should be covered.

**Comment by sliwinski-milosz** [comment]
Sorry you are right, `get_marker` triggers warning too.
So `get_closes_marker` is a better solution here.

We still access use MarkInfo objects while testing pytest-bdd. Would be good to fix that as well:
```
/Users/milosz.sliwinski/workspace/pytest-bdd/tests/feature/test_outline.py:17: RemovedInPytest4Warning: MarkInfo objects are deprecated as they contain merged marks which are hard to deal with correctly.
Please use node.get_closest_marker(name) or node.iter_markers(name).
Docs: https://docs.pytest.org/en/latest/mark.html#updating-code
  assert test_outlined.parametrize.args == (
```

**Comment by sliwinski-milosz** [comment]
Soon I'll push fix for this warning so please do not rush with `3.0.1` release as we could put this fix into it.

**Comment by Vic152** [comment]
Hey Guys I still get this warning:

pytest_bdd\reporting.py:78: RemovedInPytest4Warning: MarkInfo objects are deprecated as they contain merged marks which are hard to deal with correctly.
Please use node.get_closest_marker(name) or node.iter_markers(name).
Docs: https://docs.pytest.org/en/latest/mark.html#updating-code
  param_names = parametrize.args[0] if isinstance(parametrize.args[0], (tuple, list)) else [

**Comment by sliwinski-milosz** [comment]
We haven't yet released the fix. It is on master branch but it haven't been pushed to pypi.

Version that will have the fix is `3.0.1`.

If you want you can install and test latest github version of `pytest-bdd` by using:
```
pip install git+https://github.com/pytest-dev/pytest-bdd.git@master
```

**Comment by Vic152** [comment]
Cool thanks, please close.

**Comment by youtux** [comment]
I will push 3.0.1 to pypi as soon as https://github.com/pytest-dev/pytest-bdd/pull/267 is merged.

**Comment by Reinaard** [comment]
Hey guys, I still don't see version 3.0.1 on pypi yet but [#267](https://github.com/pytest-dev/pytest-bdd/pull/267) is merged already! Could you please publish it?

**Comment by youtux** [comment]
I just released pytest-bdd 3.0.1 on PyPI, sorry for the long wait.

**Comment by Reinaard** [comment]
Awesome, thank you!
